### PR TITLE
`pre-commit` Enhancement

### LINKED ISSUE #114: Enhancement: Make available as `pre-commit` hook
Include a `.pre-commit-hooks.yaml` file so this can be easily used in a `pre-commit` set up.

(I have an incoming PR to provide this.)

### PR DESCRIPTION
**Pull request by mindlessroman**
This PR: 
* Adds a **`.pre-commit-hooks.yaml` file** so this really handy tool can be used in `pre-commit` CI processes!
* Updates some of the **dependencies' versions** (found while debugging above).
* Updates some **failing tests** that were due to timeouts (found when confirming I wasn't breaking anything existing).
* **Updates the Changelog:** What's happening with the upcoming change.
* **Updates the README:** Added a section for the installation in a `pre-commit` style `.yaml` format.

Testing: 
* Ran it locally using the `pre-commit try-repo`: `pre-commit try-repo ../spellchecker-cli/ spellchecker-cli` ✅ 
* Ran the `npm run test` ✅ 

PR TODOS:
* [x] If you're adding or changing a command-line flag, update the flags list in [the Usage section of the README](https://github.com/tbroadley/spellchecker-cli#usage).
* [x] If you're adding a feature or changing behaviour, document the change in the README.
* [x] Add a line to the Unreleased section of CHANGELOG.md describing your change (and crediting yourself if you'd like!)

Would close #114

### CODE REVIEW

**Comment by mindlessroman** [comment]
Would close #114

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+- id: spellchecker-cli
+  name: spellchecker
+  entry: spellchecker
+  language: node
+  types: [markdown]
```
Comment (tbroadley) [inline .pre-commit-hooks.yaml:5]: Would it make sense to also spellcheck other kinds of plain text files here? Perhaps:

```suggestion
  types: [text]
```

Or

```suggestion
  types_or: [text, markdown]
```

If `markdown` doesn't count as a subtype of `text`.

File: yarn.lock [hunk 2]
```diff
@@ -3,3196 +3,3399 @@
 
 
 "@babel/code-frame@7.12.11":
-  version "7.12.11"
-  resolved "https://registry.yarnpkg.com/@babel/code-frame/-/code-frame-7.12.11.tgz#f4ad435aa263db935b8f10f2c552d23fb716a63f"
-  integrity sha512-Zt1yodBx1UcyiePMSkWnU4hPqhwq7hGi2nFL1LeA3EUl+q2LQx16MISgJ0+z7dnmgvP9QtIleuETGOiOH1RcIw==
+  "integrity" "sha512-Zt1yodBx1UcyiePMSkWnU4hPqhwq7hGi2nFL1LeA3EUl+q2LQx16MISgJ0+z7dnmgvP9QtIleuETGOiOH1RcIw=="
+  "resolved" "https://registry.npmjs.org/@babel/code-frame/-/code-frame-7.12.11.tgz"
```
Comment (tbroadley) [inline yarn.lock:7]: I noticed that the "resolved" fields on all dependencies changed from pointing to registry.yarnpkg.com to registry.npmjs.org. I'm not sure if it makes sense to make that change. If you have a reason for this change, let me know and let's do it in a separate PR! 

For this PR, would you mind looking into reverting these changes?

File: package.json [hunk 3]
```diff
@@ -65,14 +65,14 @@
...
-    "@types/mocha": "^8.2.2",
+    "@types/mocha": "^10.0.1",
     "@types/node": "^15.12.2",
     "@typescript-eslint/eslint-plugin": "^4.26.1",
     "@typescript-eslint/parser": "^4.26.1",
     "chai": "^4.3.4",
-    "eslint": "^7.28.0",
-    "eslint-config-airbnb-base": "^13.2.0",
-    "eslint-plugin-import": "^2.23.3",
+    "eslint": "^7.32.0 || ^8.2.0",
```
Comment (tbroadley) [inline package.json:73]: I wonder if we should simply upgrade to 

```suggestion
    "eslint": "^8.2.0",
```

Or is it necessary to support both v7 and v8? I don't think it is, just because `yarn.lock` will lock us into a particular version that presumably only matches one of these specifiers.

**Review by tbroadley — COMMENTED** [review]
Thank you so much for this!

File: .pre-commit-hooks.yaml (hunk 1 above)
Comment (mindlessroman) [reply .pre-commit-hooks.yaml:5]: https://github.com/pre-commit/identify/blob/main/identify/extensions.py#L141 
I think markdown is considered text! 

I think if it's widened to `text`, then the `--files` flag would have to be present if anyone is looking for specific types. My team's usecase is for `markdown` only so we'd probably tune it to `--files "**/*.md`. 

FWIW, Turning to `text` in my local test just now (`pre-commit try-repo ../spellchecker-cli/ spellchecker-cli --all-files`) grabbed `.yaml`, '.ipynb`, `.ignore`-like files, and more.

File: yarn.lock (hunk 2 above)
Comment (mindlessroman) [reply yarn.lock:7]: Absolutely! this was autogenerated, so I can definitely roll back.

File: package.json (hunk 3 above)
Comment (mindlessroman) [reply package.json:73]: Fair enough! I can try this version and make sure it doesn't fail tests

Comment (mindlessroman) [reply package.json:73]: Changing just to `"eslint": "^8.2.0",` has peer dependency failures.

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: @typescript-eslint/eslint-plugin@4.26.1
npm ERR! Found: eslint@8.47.0
npm ERR! node_modules/eslint
npm ERR!   dev eslint@"^8.2.0" from the root project
npm ERR!   peer eslint@"^6.0.0 || ^7.0.0 || >=8.0.0" from @eslint-community/eslint-utils@4.4.0
npm ERR!   node_modules/@eslint-community/eslint-utils
npm ERR!     @eslint-community/eslint-utils@"^4.2.0" from eslint@8.47.0
npm ERR!   4 more (@typescript-eslint/experimental-utils, ...)
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer eslint@"^5.0.0 || ^6.0.0 || ^7.0.0" from @typescript-eslint/eslint-plugin@4.26.1
npm ERR! node_modules/@typescript-eslint/eslint-plugin
npm ERR!   dev @typescript-eslint/eslint-plugin@"^4.26.1" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: eslint@7.32.0
npm ERR! node_modules/eslint
npm ERR!   peer eslint@"^5.0.0 || ^6.0.0 || ^7.0.0" from @typescript-eslint/eslint-plugin@4.26.1
npm ERR!   node_modules/@typescript-eslint/eslint-plugin
npm ERR!     dev @typescript-eslint/eslint-plugin@"^4.26.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

File: yarn.lock (hunk 2 above)
Comment (mindlessroman) [reply yarn.lock]: Done!

File: .pre-commit-hooks.yaml (hunk 1 above)
Comment (GeisaFaustino) [reply .pre-commit-hooks.yaml:5]: I was just going to ask about jupyter notebooks and comments (docstring) in python scripts... 🤣
Does it works for python too?

**Comment by GeisaFaustino** [comment]
Great job @mindlessroman 👏👏👏
Thank you so much for making our lives easier 🤣

**Review by tbroadley — COMMENTED** [review]
Hmm I would expect there to be some changes to `yarn.lock`, since we're upgrading dependencies. If you like, I can pull down this PR to my local machine and try to resolve that.

**Comment by mindlessroman** [comment]
> Hmm I would expect there to be some changes to `yarn.lock`, since we're upgrading dependencies. If you like, I can pull down this PR to my local machine and try to resolve that.

That'd be great! I don't usually use yarn so maybe there's some goof I made along the way.

File: package.json (hunk 3 above)
Comment (mindlessroman) [reply package.json:73]: @tbroadley - I think it may make sense (or be less noisy) to support both for now?

Comment (mindlessroman) [reply package.json:73]: Going back to `"eslint": "^7.32.0 || ^8.2.0",` allowed me to `npm install` happily.

Comment (mindlessroman) [reply package.json:73]: I'll resolve this for now.
